### DISCUSSION TITLE
Enhancement: Custom Endpoint timeout.

### DISCUSSION THREAD

**Discussion by Originalimoc** [danny-avila/LibreChat#4401] [Feature Requests & Suggestions]
### What features would you like to see added?

Sometimes the API will return data from MoA backend then the time might exceed 30s even 1 minutes+ per request.

### More details

Add a timeout and stream mode off to custom endpoints.

### Which components are impacted by your request?

Endpoints

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
There is no timeout configuration, are you running LibreChat in front of NGINX/traefik? In which case, you can set a timeout within those settings, as defaults there are usually 1 minute.

> stream mode off to custom endpoints.

already implemented. Add `stream` under `dropParams`:

https://www.librechat.ai/docs/configuration/librechat_yaml/object_structure/custom_endpoint#dropparams

**Reply by Manouchehri** [reply]
Isn't it defined in `requestOptions`?  https://github.com/Azure/azure-sdk-for-js/issues/26243#issuecomment-1693955243

**Reply by OrimoB** [reply]
> Isn't it defined in `requestOptions`? [Azure/azure-sdk-for-js#26243 (comment)](https://github.com/Azure/azure-sdk-for-js/issues/26243#issuecomment-1693955243)

That's on API server side... not LibreChat server side
